airflow/utils/helpers.py

```diff
@@ -205,10 +205,16 @@ def kill_descendant_processes(logger, pids_to_kill=None):
     logger.warn("Terminating descendant processes of {} PID: {}"
                 .format(this_process.cmdline(),
                         this_process.pid))
-    for descendant in descendant_processes:
+
+    temp_processes = descendant_processes
```

Comment:
@bolkedebruin just played around with some code an I think you really want to make a copy with:

`temp_processes = descendant_processes[:]

If you fix it, I will merge the code in (I can now :-))
`